core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/adapter/AbstractServiceProviderAdapter.java

```diff
@@ -0,0 +1,18 @@
+package com.sequenceiq.cloudbreak.service.stack.connector.adapter;
+
+
+import com.google.common.base.Optional;
+import com.sequenceiq.cloudbreak.cloud.event.context.CloudContextParams;
+import com.sequenceiq.cloudbreak.domain.FailurePolicy;
+import com.sequenceiq.cloudbreak.domain.Stack;
+
+abstract class AbstractServiceProviderAdapter {
```

Comment:
Remove this. We are using converters. e.g. com.sequenceiq.cloudbreak.converter.spi.StackToCloudStackConverter.java. If you need a conversion between two objects e.g. Credential to CloudContext or Stack to CloudContext then create a converter.